File: Java/src/test/java/com/gildedrose/types/SulfurasTest.java
package com.gildedrose.types;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

import com.gildedrose.Item;
import com.gildedrose.enums.ProductType;

public class SulfurasTest {
	
	@Test
	public void testSulfurasUpdateQuality() {
		Item item = new Item(ProductType.SULFURAS.name(), -1, 80);
		Sulfuras sulfuras = new Sulfuras();
		sulfuras.updateQuality(item);
		assertEquals(item.quality, 80);
		assertEquals(item.sellIn, -1);
	}

}

File: Kotlin/src/test/kotlin/com/platinumrose/advance/quality/SulfurasTest.kt
package com.platinumrose.advance.quality

import com.platinumrose.ItemConstant
import com.platinumrose.ItemType
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import kotlin.random.Random

class SulfurasTest {

    private val Random = Random(Int.MAX_VALUE)
    private val item = Sulfuras()

    @Test
    fun `should return expected type`() {
        assertEquals(item.type(), ItemType.SULFURAS)
    }

    @Test
    fun `should return expected max quality`() {
        assertEquals(item.maxQuality(), ItemConstant.LEGENDARY_ITEM_MAX_QUALITY)
    }

    @Test
    fun `should return 0 when calculate sell in decrease`() {
        assertEquals(item.computeSellInDecrease(Random.nextInt()), 0)
    }

    @Test
    fun `should return 0 when calculate quality increase before sell in`() {
        assertEquals(item.computeQualityIncreaseBeforeSellIn(Random.nextInt(), Random.nextInt()), 0)
    }

    @Test
    fun `should return 0 when calculate quality increase after sell in`() {
        assertEquals(item.computeQualityIncreaseAfterSellIn(Random.nextInt(), Random.nextInt()), 0)
    }
}